glossary.ftd

```diff
@@ -1,22 +1,219 @@
+processor.
+
+
+
+-- ft.h2: document-suffix
+
+The `document-suffix` is the special read-only property in
+[`document-full-id`](glossary/#document-full-id) which is added after special
+character `/-/`.
+
+It helps to provide different [`document-metadata`](glossary/#document-metadata)
+to render different content.
+
+For `/foo/-/x/` as `document-full-id`, /x/ is `document-suffix`.
+
+This can accessed using [`document-suffix`](processors/document-suffix/)
+processor.
+
+
+-- ft.h2: file
+
+The unit of storage in FPM. It is uniquely identified by the
+[filename](glossary/#filename).
+
+
+
+-- ft.h2: filename
+
+The unique identifier of an [file](glossary/#file).
+
+
+
+-- ft.h2: fpm project
+
+The unit folder that contains `FPM.ftd` and any number of folders or files of
+any extension along with states (server-state and/or client-state).
+
+
+-- ft.h2: server project
+
+An fpm project which is used to track the same project but resides somewhere
+else. To communicate with server, see [clone](glossary/#clone) or
+[sync](glossary/#sync).
+
+
+
+-- ft.h2: server-latest
+
+The current tree of files in server. This can be derived by looking at the server
+history and get the latest version of each file. The file having the content
+from its latest version, if not deleted, becomes the part of current tree.
+
+
+-- ft.h2: sync
+
+To bring the contents of [client project](glossary/#client-project) into
+the [server project](glossary/#server-project). Also to bring the server's
+changes after the client diverged into the client project. In the end results
+the server and the client to be exact copy of each other.
+
+Sync is specific to a file. Though the [`fpm sync`](cli/sync/) commands
+results in syncing of all the changed file.
+
+Syncing is performed by an automatic process that identifies changes made since
+the client and the server diverged, and then applies all those changes together.
+In cases where changes conflict in the file, file enters into conflict-state and
+manual intervention may be required to complete the sync of conflicted file.
+
+A successful sync results in the creation of a new [version](glossary/#version)
+of the file.
+
+
+-- ft.h2: three-way merge
+
+Three-way merges use a dedicated [version](glossary/#version), also known
+as base version, to tie together the histories in server and [un-synced
+file](glossary/#un-synced-file) in client. FPM uses three state of file to
+generate the new version: first: latest version in server, second: un-synced
+file in client and third: their common ancestor version.
+
+
+-- ft.h2: tracked file
+
+The file that resides in the [client project](glossary/#client-project) and
+has entry in workspace.
+
+
+-- ft.h2: un-synced file
+
+The changes in the file that resides in the [client
+project](glossary/#client-project) and are not yet [sync](glossary/#sync)ed.
+
+
+-- ft.h2: un-tracked file
+
+The file that resides in the [client project](glossary/#client-project) but
+has no entry in workspace.
```

Comment:
But is not part of package. 

Ideally we should change our `fpm serve`, `import` etc behaviour also to not touch these files.